var/spack/repos/builtin/packages/qt-base/package.py

```diff
@@ -0,0 +1,112 @@
+    variant('gtk', default=False, description='Build with gtkplus.')
+    variant('gui', default=True, description='Build the Qt GUI module and dependencies.')
+    variant('opengl', default=False, description='Build with OpenGL support.')
+    variant('shared', default=True, description='Build shared libraries.')
+    variant('sql', default=True, description='Build with SQL support.')
+    variant('ssl', default=True, description='Build with OpenSSL support.')
+    variant('tests', default=False, description='Build tests.')
+    variant('widgets', default=True, description='Build with widgets.')
+
+    depends_on('cmake@3.16:', type='build')
+    depends_on('ninja', type='build')
+    depends_on("pkgconfig", type='build')
+    depends_on("python", type='build')
+
+    # Dependencies, then variant- and version-specific dependencies
+    depends_on('at-spi2-core', when='+accessibility')
+    depends_on('dbus', when='+dbus')
+    depends_on('double-conversion')
+    depends_on('fontconfig')
+    depends_on("freetype")
+    depends_on('gl', when='+opengl')
+    depends_on("harfbuzz")
+    depends_on("icu4c")
+    depends_on("jpeg")
+    depends_on('libdrm')
+    depends_on('libjpeg')
+    depends_on("libmng")
+    depends_on('libproxy')
+    depends_on("libtiff")
+    depends_on('libxkbcommon')
+    depends_on("libxml2")
+    depends_on('libxrender')
+    depends_on('openssl', when='+ssl')
+    depends_on('pcre2+multibyte')
+    depends_on("sqlite", when='+sql')
+    depends_on("zlib")
+    depends_on("zstd")
+
+    def patch(self):
+        import os
+        import shutil
+        vendor_dir = join_path(self.stage.source_path, 'src/3rdparty')
+        vendor_deps_to_keep = [
+            'blake2', 'easing', 'forkfd', 'freebsd',
+            'icc', 'md4', 'md4c', 'md5', 'rfc6234',
+            'sha1', 'sha3', 'tinycbor', 'VulkanMemoryAllocator',
+        ]
+        with working_dir(vendor_dir):
+            for dep in os.listdir():
+                if os.path.isdir(dep):
+                    if dep not in vendor_deps_to_keep:
+                        shutil.rmtree(dep)
+
+    def cmake_args(self):
+        args = [
+            self.define('FEATURE_optimize_size',
+                        self.spec.satisfies('build_type=MinSizeRel')),
+            self.define_from_variant('FEATURE_accessibility', 'accessibility'),
+            self.define_from_variant('FEATURE_dbus', 'dbus'),
+            self.define('INPUT_dbus', 'linked'),
+            self.define_from_variant('FEATURE_framework', 'framework'),
+            self.define_from_variant('FEATURE_gui', 'gui'),
+            self.define_from_variant('FEATURE_ssl', 'ssl'),
+            self.define('INPUT_openssl', 'linked'),
+            self.define_from_variant('FEATURE_widgets', 'widgets'),
+            self.define_from_variant('QT_BUILD_EXAMPLES', 'examples'),
+            self.define_from_variant('QT_BUILD_TESTS', 'tests'),
```

Comment:
I don't think we want a `test` variant since it presumably doesn't change the installed binaries. Likewise, if the examples aren't installed they shouldn't get a variant.

Reply:
I'll change that to use `self.run_tests` on build.